tools/src/launcher/Main.hs

```diff
@@ -220,38 +228,58 @@ getLauncherOptions = do
 
     -- Poor man's environment variable expansion.
     expandVars :: LauncherOptions -> IO LauncherOptions
-#ifdef mingw32_HOST_OS
-    expandVars lo@(LO {..}) = do
+    expandVars lo@(LO {..}) = case buildOS of 
+        Windows -> do
         -- %APPDATA%: nodeArgs, nodeDbPath,
         --     nodeLogPath, updaterPath,
         --     updateWindowsRunner, launcherLogsPrefix
         -- %DAEDALUS_DIR%: nodePath, walletPath
-        appdata <- toText <$> getEnv "APPDATA"
-        daedalusDir <- (toText . takeDirectory) <$> getExecutablePath
-        let replaceAppdata = replace "%APPDATA%" appdata
-            replaceDaedalusDir = replace "%DAEDALUS_DIR%" daedalusDir
-        pure lo
-            { loNodeArgs            = map (T.replace "%APPDATA%" appdata) loNodeArgs
-            , loNodeDbPath          = replaceAppdata loNodeDbPath
-            , loNodeLogPath         = replaceAppdata <$> loNodeLogPath
-            , loUpdaterPath         = replaceAppdata loUpdaterPath
-            , loUpdateWindowsRunner = replaceAppdata <$> loUpdateWindowsRunner
-            , loLauncherLogsPrefix  = replaceAppdata <$> loLauncherLogsPrefix
-            , loNodePath            = replaceDaedalusDir loNodePath
-            , loWalletPath          = replaceDaedalusDir <$> loWalletPath
-            }
-#else
-    expandVars lo@(LO {..}) = do
-        home <- toText <$> getEnv "HOME"
-        let replaceHome = replace "$HOME" home
-        pure lo
-            { loNodeArgs           = map (T.replace "$HOME" home) loNodeArgs
-            , loNodeDbPath         = replaceHome loNodeDbPath
-            , loNodeLogPath        = replaceHome <$> loNodeLogPath
-            , loUpdateArchive      = replaceHome <$> loUpdateArchive
-            , loLauncherLogsPrefix = replaceHome <$> loLauncherLogsPrefix
-            }
-#endif
+            appdata <- toText <$> getEnv "APPDATA"
+            daedalusDir <- (toText . takeDirectory) <$> getExecutablePath
+            let replaceDaedalusDir = replace "%DAEDALUS_DIR%" daedalusDir
+
+                replaceAppdata = replace "%APPDATA%" appdata 
+                replaceUpdaterAppdata (WindowsUpdaterData wud@WinUD {..}) = WindowsUpdaterData wud 
+                    { wudExePath = replaceAppdata wudExePath
+                    , wudBatPath = replaceAppdata wudBatPath
+                    }
+                replaceUpdaterAppdata _ = error "Can not expand %APPDATA%"
```

Comment:
No, don't use `error` here please, as I said earlier `error` means a programmer mistake, but here we don't which options will be passed.